src/settings.c

```diff
@@ -29,16 +28,8 @@ static FILE *xdg_config(const char *filename)
         f = fopen(path, "r");
         g_free(path);
 
-        for (const gchar * const *d = systemdirs;
-             !f && *d;
-             d++) {
-                path = g_strconcat(*d, filename, NULL);
-                f = fopen(path, "r");
-                g_free(path);
-        }
-
         if (!f) {
-                f = fopen("/etc/dunst/dunstrc", "r");
+                f = fopen(SYSCONFFILE, "r");
```

Comment:
> On the one hand you want Dunst to look at the conf dir you said to put the config file in, _but on the other hand it might confuse users that otherwise identical dunst installs behave differently._

Not this one right here. ;) That is exactly what I'd expect, because local software should stay out of everything that does not belong to its prefix. Plus, different behaviour should be expected anyway since one installs almost certainly a more up-to-date version from upstream, which could always involve deprecation of config settings. If users do want to share the config, they can just make $PREFIX/etc/xdg/dunst a symlink to /etc/xdg/dunst or the other way around.

As examples, again: mpv and unbound, which I happen to have installed locally and which make actual use of $PREFIX/etc. They even incorporate the prefix in the documentation.

> I like the env change better, since we can stick to a standard way to look up the config and users have the flexibility to decide what they want. 

But that is *less* user-friendly since it involves changing their environment, which I am not so sure anymore is/was the right thing to do in the first place. The way XDG_CONFIG_DIRS works and the current way dunst makes use of it, leads to situations where the user does not know, which config gets used. The default on Ubuntu is an unset XDG_CONFIG_DIRS, which in turn defaults to `/etc/xdg`, which is precisely what I *don't* want. Even if I set it up like this:
```sh
XDG_CONFIG_DIRS=/usr/local/etc/xdg:/etc/xdg
```
globally, i.e. in /etc/environment, it does not work as intended, because then any distro software which is 1. installed and 2. has also been installed locally, will then prefer /usr/local/etc, which is definitely wrong. There should be as much isolation between the two prefixes as possible, at least by default. Advanced users with special needs should not have a problem with making it work.
I also think now that g_get_system_config_dirs() has no use in regular software for precisely those reasons just outlined. It may have its use for programs that update those directories or whatever. If the authors intended it differently, well, they have left a huge gaping hole in the spec as to how to deal with locally installed software that does not lead to funny behaviour. ;) Maybe the intended use is to prepend PREFIX, when one queries XDG_CONFIG_DIRS. That could be a solution which would result in what I expect, but it is undocumented, and one should never rely on undocumented behaviour. And XDG_CONFIG_DIRS would then need to be queried at configure/compile time. Since it is not a simple string but an array split by ':', I would not want to deal with that if I can help it. I mean, checking it, possibly having to default, and if it has an actual value, splitting and walking it, seems like a lot of hassle, when the default on distros is unset anyway. I would sooner risk a user complaint about an issue in a rather distant future, and fix it then, than doing this stuff now, while the correct thing to do is still unclear. Anyone that runs a distro which actually does set this parameter, please come forward now. ;-)

> We should just instruct users to change their global environment when they want dunst to look at the config. For most users it won't be a problem anyways, since they make a user config that overrides the system one.

As I just outlined, that might be a bad idea, especially if they use a user config, because they might also install other software locally, keeping the distro version installed, not have a user config for it, and then all hell breaks loose: "Dunst broke my toys!!1!", or whatever, but not before they go running and complain to the distro maintainers. ;)
I think, for the time being, this solution is dead simple and should do the right thing in most cases. I can just make the previous code a comment, so it is still there, should it be necessary to reactivate it. But then, what is git for? ;)

Reply:
Sorry I forgot that we actually expect different behaviour from the two.

Reply:
Oh wait, I need to roll back the other changes which set XDG_CONFIG_DIRS. They should not hurt, since they don't do anything now, but they are unnecessary an might confuse people that take a look...

Reply:
One more thing that did get pointed out to me on the XDG mailing list: The spec talks about "information" as opposed to files, so just because the one file in a more important location (i.e. the first) does not mean, you should not try to find *other* information in less important locations. So, technically dunst does not do that either, and if it would do that with i.e. `XDG_CONFIG_DIRS=/usr/local/etc/xdg:/etc/xdg`, then it can get really messy: Read setting A from /usr/local/etc/xdg... but since setting B is not there move on and read it from /etc/xdg... where it is set. That situation could also arise and would be the same for locally installed and distro software.

See the [spec](https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html) for reference:
> When the same **information** is defined in multiple places the information defined relative to the more important base directory takes precedent.

This means that the current way of dealing with configs is not technically correct, because it uses **either** the user config, if found, **or** the system-wide config otherwise. But this has nothing to do directly with this PR so I won't do anything about it. Just wanted to add this bit, since it bares *some* relevance.

Reply:
Good to know. This isn't really high priority, since I want to keep the systemd wide dunstrc the same as the in-code defaults. So in practice it would only make a difference if one changes their system-wide config.
Just for reference, [this](https://lists.freedesktop.org/archives/xdg/2021-September/014560.html) is the mentioned email thread

Reply:
From the emails:

> HOWEVER, Base Directory does not specify a first match wins rule.  It
> attributes more importance (the spec's terminology) to files located
> in earlier directories in the list, but that does not imply that only
> one can be used.

From the spec:

> When the same information is defined in multiple places the information defined relative to the more important base directory takes precedent.

This doesn't sound like you *may not* use a first match wins rule.

Reply:
> Just for reference, [this](https://lists.freedesktop.org/archives/xdg/2021-September/014560.html) is the mentioned email thread

Correct, should have linked it myself, sorry.

> This doesn't sound like you *may not* use a first match wins rule.

Yes, I think you are right, but it would be nice to have. Basically I think this could be done quite easily, though. If you look at shell initialization, for instance: Read the *least* important files first and work your way to the most important ones, or simply concatenate the files in that order and just overwrite values as they come. Since the last value comes from the most important file, that's it.

Haven't looked how much work that would take to change in dunst's code, though. I am holding off on that for now.